src/main/java/com/aws/greengrass/lifecyclemanager/GreengrassService.java

```diff
@@ -452,8 +456,10 @@ protected CompletableFuture<Void> close(boolean waitForDependers) {
                         logger.error("Interrupted waiting for dependers to exit");
                     }
                 }
+                // removing listeners on dependencies after the dependers have exited
+                dependencies.forEach((service, dependencyInfo) ->
+                        getContext().removeGlobalStateChangeListener(dependencyInfo.stateListener));
                 externalDependenciesTopic.remove(externalDependenciesTopicWatcher);
-                lifecycle.setClosed(true);
                 requestStop();
```

Comment:
should we close and requeststop immediately?

Reply:
ideally, maybe yes, but since it will be a incompatible behavioral change, I don't think we should so it now.